tools/src/main/java/org/apache/kafka/tools/BrokerApiVersionsCommand.java

```diff
@@ -0,0 +1,337 @@
+                    metrics,
+                    time,
+                    "admin",
+                    ClientUtils.createChannelBuilder(config, time, logContext),
+                    logContext);
+            NetworkClient networkClient = new NetworkClient(
+                    selector,
+                    metadata,
+                    clientId,
+                    DEFAULT_MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION,
+                    DEFAULT_RECONNECT_BACKOFF_MS,
+                    DEFAULT_RECONNECT_BACKOFF_MAX,
+                    DEFAULT_SEND_BUFFER_BYTES,
+                    DEFAULT_RECEIVE_BUFFER_BYTES,
+                    config.getInt(CommonClientConfigs.REQUEST_TIMEOUT_MS_CONFIG),
+                    config.getLong(CommonClientConfigs.SOCKET_CONNECTION_SETUP_TIMEOUT_MS_CONFIG),
+                    config.getLong(CommonClientConfigs.SOCKET_CONNECTION_SETUP_TIMEOUT_MAX_MS_CONFIG),
+                    time,
+                    true,
+                    new ApiVersions(),
+                    logContext,
+                    MetadataRecoveryStrategy.NONE);
+            ConsumerNetworkClient highLevelClient = new ConsumerNetworkClient(
+                    logContext,
+                    networkClient,
+                    metadata,
+                    time,
+                    config.getLong(CommonClientConfigs.RETRY_BACKOFF_MS_CONFIG),
+                    config.getInt(CommonClientConfigs.REQUEST_TIMEOUT_MS_CONFIG),
+                    Integer.MAX_VALUE);
+            return new AdminClient(time, highLevelClient, metadata.fetch().nodes());
+        }
+
+        AdminClient(Time time, ConsumerNetworkClient client, List<Node> bootstrapBrokers) {
+            this.time = time;
+            this.client = client;
+            this.bootstrapBrokers = bootstrapBrokers;
+            startNetworkThread();
+        }
+
+        private void startNetworkThread() {
+            Thread networkThread = new Thread(() -> {
+                try {
+                    while (running) {
+                        client.poll(time.timer(Long.MAX_VALUE));
+                    }
+                } catch (Throwable t) {
+                    LOGGER.error("admin-client-network-thread exited", t);
+                } finally {
+                    pendingFutures.forEach(future -> {
+                        try {
+                            future.raise(Errors.UNKNOWN_SERVER_ERROR);
+                        } catch (IllegalStateException ignored) {
+                        }
+                    });
+                    pendingFutures.clear();
+                }
+            }, "admin-client-network-thread");
+            networkThread.setDaemon(true);
+            networkThread.start();
+        }
+
+        private AbstractResponse send(Node target, AbstractRequest.Builder<?> request) throws InterruptedException {
+            RequestFuture<ClientResponse> future = client.send(target, request);
+            pendingFutures.add(future);
```

Comment:
Do we need another thread to poll client? Maybe we can poll the client by the same thread to simplify this tool?